File: articles/analytical-psychology-and-clinical-practice/go/clinical_functioning.go
package main

import "fmt"

func compensatoryPressure(consciousOnesidedness, complexActivation, traumaFragmentation, egoIntegration, shadowAwareness float64) float64 {
	return 0.62*consciousOnesidedness +
		0.58*complexActivation +
		0.34*traumaFragmentation -
		0.52*egoIntegration -
		0.26*shadowAwareness
}

func clinicalFunctioning(symptomBurden, egoIntegration, symbolicCapacity, relationalSafety, affectRegulation, compensatory, shadowAwareness float64) float64 {
	return -0.70*symptomBurden +
		0.60*egoIntegration +
		0.52*symbolicCapacity +
		0.64*relationalSafety -
		0.42*compensatory +
		0.28*affectRegulation +
		0.24*shadowAwareness
}

func main() {
	pressure := compensatoryPressure(0.36, 0.42, 0.24, 0.80, 0.80)
	score := clinicalFunctioning(0.38, 0.80, 0.76, 0.82, 0.78, pressure, 0.80)

	fmt.Printf("Synthetic compensatory pressure: %.3f\n", pressure)
	fmt.Printf("Synthetic clinical-functioning score: %.3f\n", score)
}

File: articles/analytical-psychology-and-clinical-practice/fortran/clinical_functioning.f90
! Analytical Psychology and Clinical Practice
! Fortran example: simple synthetic clinical-functioning score

program clinical_functioning_demo
  implicit none

  real :: pressure
  real :: score

  pressure = 0.62 * 0.36 + 0.58 * 0.42 + 0.34 * 0.24 - &
             0.52 * 0.80 - 0.26 * 0.80

  score = -0.70 * 0.38 + 0.60 * 0.80 + 0.52 * 0.76 + &
          0.64 * 0.82 - 0.42 * pressure + 0.28 * 0.78 + &
          0.24 * 0.80

  print *, "Synthetic compensatory pressure:", pressure
  print *, "Synthetic clinical-functioning score:", score
end program clinical_functioning_demo
